src/lib/http/http_server.pl

```diff
@@ -110,6 +108,15 @@
 
 path([]) --> [].
 
+send_response(Stream, http_response(StatusCode0, file(Filename), Headers)) :-
+    default(StatusCode0, 200, StatusCode),
+    format(Stream, "HTTP/1.0 ~d\r\n", [StatusCode]),
+    overwrite_header("Connection"-"Close", Headers0, Headers1),
+    write_headers(Stream, Headers1),
+    format(Stream, "\r\n", []),
+    open(Filename, read, FileStream, [type(binary)]),
```

Comment:
`open/[3,4]` is best used in connection with `setup_call_cleanup/3` to reliably close the stream.

For instance, in this case:

<pre>
setup_call_cleanup(open(FileName, read, FileStream, [type(binary)]), 
                   pipe_bytes(FileStream,Stream),
                   close(FileStream))
</pre>